src/features/lti-resources/ResourceLinkGenerator.jsx

```diff
@@ -0,0 +1,90 @@
+import React, { useState } from 'react';
+import { useSelector } from 'react-redux';
+import PropTypes from 'prop-types';
+import { Button, ModalDialog, useToggle } from '@edx/paragon';
+import { currentSequenceSelector } from '../course-player/data/selectors';
+
+// Used to generate resource identifiers for OEX as a LTI consumer
+const ResourceLinkGenerator = ({
+  sequenceId,
+  courseId,
+  unitId,
+}) => {
+  const currentSequence = useSelector(currentSequenceSelector);
+  const currentSectionId = currentSequence ? currentSequence.sectionId : null;
+
+  const generateCourseResource = () => {
```

Comment:
if this funtion doesn't need to have context of the component you can move it outside of the component and make it pure so it's easier to test